Bluetooth: Mesh: Add CDB depend on to prov Kconfig

Andrewpini opened the pull request:
The current implementation of provisioning features is dependent on CDB
to compile, which is reflected poorly in the Kconfig options for mesh.

This commit alters the Kconfig options for provisioning to be dependent
on the CDB option to compile.

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>

PavelVPV commented:
Minor: Please use imperative mood in the subject line of the commit with some explanation.

Andrewpini committed:
Bluetooth: Mesh: Add CDB depend on to prov Kconfig

The current implementation of provisioning features is dependent on CDB
to compile, which is reflected poorly in the Kconfig options for mesh.

This commit alters the Kconfig options for provisioning to be dependent
on the CDB option to compile.

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>
